File: CodeWars/Python/7 kyu/Linked Lists - Push & BuildOneTwoThree/Source.py
class Node(object):
    def __init__(self, data):
        self.data = data
        self.next = None
    
def push(head, data):
    tmp = Node(data)
    tmp.next = head
    return tmp
  
def build_one_two_three():
    res = None
    for x in range(3, 0, -1):
        res = push(res, x)
        
    return res

File: CodeWars/C++/6 kyu/Linked Lists - Length & Count/Source.cpp
/* Node Definition
struct Node {
  Node * next;
  int data;
}
*/

int Length(Node *head)
{
  auto lenght = 0;
  while(head) 
  {
    ++lenght;
    head = head->next;
  }
  return lenght;
}

int Count(Node *head, int data)
{
  auto count = 0;
  while(head) 
  {
    if(head->data == data) ++count;
    head = head->next;
  }
  return count;
}